### DISCUSSION TITLE
Config as Code / Gitops possible?

### DISCUSSION THREAD

**Discussion by refactoriel** [zitadel/zitadel#6963] [Q&A] [answered]
Hi,

I'm currently evaluating Zitadel as an alternative to Authentik.

AFAICS it used to be possible to run Zitadel in "gitops mode": https://github.com/zitadel/zitadel/blob/v1.87.5/docs/docs/guides/installation/gitops.md

Is something similar still possible? I'd like to be able to configure everything without using the GUI.

**Comment by stebenz** [comment, marked as answer]
Hi @DanBenHa,
GitOps mode was in this old version used to deploy ZITADEL itself in a Kubernetes cluster, to self-host I would direct you to https://github.com/zitadel/zitadel-charts which houses our helm chart. This helm chart could be used with for example ArgoCD to achieve GitOps for your own infrastructure.
If it's for the resources in ZITADEL, you could use terraform(https://registry.terraform.io/providers/zitadel/zitadel/latest) which should also be possible in a similar fashion to GitOps.

I hope this helps you.

**Reply by refactoriel** [reply]
Awesome, thank you! I was looking for a way to both configure infrastructure and Zitadel resources via Gitops. In my case, the latter could be achieved using fluxcd's terraform controller with the Zitadel terraform provider.

~~On more question though. Is it possible to disable the initial password reset that is required for the admin user? `ZITADEL_FIRSTINSTANCE_ORG_HUMAN_PASSWORD` sets the initial password, but `ZITADEL_DEFAULTINSTANCE_LOGINPOLICY_HIDEPASSWORDRESET` does not disable the password reset.~~ Nevermind, I think I just found it 1 minute after asking: https://github.com/zitadel/zitadel/blob/main/cmd/setup/steps.yaml#L32
